### ISSUE TITLE
CI: asr-template_add_01b failure

### ISSUE THREAD

**Issue by certik** [lfortran/lfortran#3736]
https://github.com/lfortran/lfortran/actions/runs/8471094950/job/23210317839?pr=3711

```
../integration_tests/template_03b.f90 * asr ✓ 
../integration_tests/template_02.f90 * asr ✓ 
../integration_tests/template_struct_01.f90 * asr ✓ 
../integration_tests/template_04.f90 * asr ✓ 
Traceback (most recent call last):
  File "/home/runner/work/lfortran/lfortran/./run_tests.py", line 496, in <module>
    tester_main("LFortran", single_test)
  File "/home/runner/work/lfortran/lfortran/src/libasr/compiler_tester/tester.py", line 479, in tester_main
    for f in futures:
  File "/home/runner/micromamba/envs/lf/lib/python3.10/concurrent/futures/_base.py", line 621, in result_iterator
    yield _result_or_cancel(fs.pop())
  File "/home/runner/micromamba/envs/lf/lib/python3.10/concurrent/futures/_base.py", line 319, in _result_or_cancel
    return fut.result(timeout)
  File "/home/runner/micromamba/envs/lf/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/home/runner/micromamba/envs/lf/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/home/runner/micromamba/envs/lf/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/runner/work/lfortran/lfortran/./run_tests.py", line 223, in single_test
    run_test(
  File "/home/runner/work/lfortran/lfortran/src/libasr/compiler_tester/tester.py", line 356, in run_test
    raise RunException(
compiler_tester.tester.RunException: Testing with reference output failed.
../integration_tests/template_add_01b.f90 * asr
The JSON metadata differs against reference results
Reference JSON: tests/reference/asr-template_add_01b-bd911f4.json
Output JSON:    tests/output/asr-template_add_01b-bd911f4.json
Omitting 10 identical items
Differing items:
{'stdout_hash': 'd14a028c2a3a2bc9476102bb288234c415a2b01f828ea62ac5b3e42f'} != {'stdout_hash': '23522768019fb1810c0e5a2e1f99e58eaa47960ca718a2f074b932b8'}
Diff against: tests/reference/asr-template_add_01b-bd911f4.stdout
```

**Comment by Shaikh-Ubaid** [comment]
It seems like a false failure to me.

**Comment by certik** [comment]
This might be similar:
```
../integration_tests/interface_02.f90 * ast ✓ 
../integration_tests/interface_03.f90 * ast ✓ 
write6.f90 * run ✓ 
../integration_tests/interface_01.f90 * ast_f90 ✓ 
builtin1.f90 * ast_f90 ✓ 
../integration_tests/interface_01.f90 * asr ✓ 
Traceback (most recent call last):
  File "/home/runner/work/lfortran/lfortran/./run_tests.py", line 496, in <module>
    tester_main("LFortran", single_test)
  File "/home/runner/work/lfortran/lfortran/src/libasr/compiler_tester/tester.py", line 479, in tester_main
    for f in futures:
  File "/home/runner/micromamba/envs/lf/lib/python3.10/concurrent/futures/_base.py", line 621, in result_iterator
    yield _result_or_cancel(fs.pop())
  File "/home/runner/micromamba/envs/lf/lib/python3.10/concurrent/futures/_base.py", line 319, in _result_or_cancel
    return fut.result(timeout)
  File "/home/runner/micromamba/envs/lf/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/home/runner/micromamba/envs/lf/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/home/runner/micromamba/envs/lf/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/runner/work/lfortran/lfortran/./run_tests.py", line 223, in single_test
    run_test(
  File "/home/runner/work/lfortran/lfortran/src/libasr/compiler_tester/tester.py", line 317, in run_test
    do = json.load(open(jo))
  File "/home/runner/micromamba/envs/lf/lib/python3.10/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/home/runner/micromamba/envs/lf/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/home/runner/micromamba/envs/lf/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/runner/micromamba/envs/lf/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

https://github.com/lfortran/lfortran/actions/runs/8470320112/job/23207764775

**Comment by certik** [comment]
Fixed by #4150.
